File: #Notion/Code/3. Sort/selection.c
#include <stdio.h>

int array[10] = {1, 10, 6, 4, 8, 2, 9, 3, 7, 5};

void printArray(int data[], int n) {
    int i;

    for (i = 0; i < n; i++)
        printf("%d ", data[i]);
    printf("\n");
}

int main(void) {
    int i, j, min, index, temp;
    int n = sizeof(array) / sizeof(array[0]);

    printArray(array, n);

    for (i = 0; i < n-1; i++) {
        min = array[i];
        index = i;

        for (j = i; j < n; j++) {
            if (min > array[j]) {
                min = array[j];
                index = j;
            }
        }
        temp = array[i];
        array[i] = array[index];
        array[index] = temp;
    }

    printArray(array, n);

    return 0;
}

File: #Notion/Code/3. Sort/Selection.java
public class Selection {
    private static int[] data = { 8, 4, 3, 7, 9, 5, 1 };

    private static void minSelectionSort(int[] data, int n) {
        int min, temp;

        for (int i = 0; i < n - 1; i++) {
            min = i;
            for (int j = i + 1; j < n; j++) {
                if (data[min] > data[j])
                    min = j;
            }
            temp = data[i];
            data[i] = data[min];
            data[min] = temp;
        }
    }

    private static void maxSelectionSort(int[] data, int n) {
        int max, temp;

        for (int i = n - 1; i > 0; i--) {
            max = i;
            for (int j = i - 1; j >= 0; j--) {
                if (data[max] < data[j])
                    max = j;
            }
            temp = data[i];
            data[i] = data[max];
            data[max] = temp;
        }
    }

    private static void printArray(int[] data) {
        for (int d : data)
            System.out.print(d + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        printArray(data);
        maxSelectionSort(data, data.length);
        printArray(data);
    }

}